---
title: Methods
layout: page
layout_key:
  -
post_slider_check_key:
  - 0
---

Please let if you find any errors, or have suggestions on how we can improve the protocols below.

## Laboratory protocols

### DNA extraction

  * [DNA extraction using silica magnetic beads][1]

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Easily automatable procedure for DNA extraction.

  * [High molecular weight DNA extraction][2]

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Useful for large-insert libraries for *de novo* genome sequencing.

* * *

### Sequencing library preparation

  * [EcoRI RAD-tag using degraded DNA (v 1.0)][3]

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This protocol can be used for degraded DNA

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Citation: [Tin *et al.* PLoS ONE 2014,Â 9(5): e96793][4]

  * [Genomic shotgun sequencing using degraded DNA][5]

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Related to the EcoRI RAD-tag protocol above, but used for whole genome shotgun sequencing.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Citation: [Tin *et al.* PLoS ONE 2014, 9(5): e96793][4]

* [Version 2 PCR free genomic shotgun sequencing using degraded DNA][9]

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Modification of original RAD-tag protocol to reduce error and bias due to PCR amplification.
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Citation: [Mikheyev *et al.* Nature Communications, 6:7991](http://www.nature.com/ncomms/2015/150806/ncomms8991/full/ncomms8991.html)

  * [RNA-seq library preparation from total RNA][6]

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Citation: [Aird *et al.* BMC Genomics 2013, 14:790][7]

   * [FASSST ddRAD tag library preparation](  * [FASSST ddRAD tag library preparation](https://docs.google.com/document/d/1uw31pcep8gjC7HnbzRAN2llhYRUVPoToD1EqSH779OA/edit?usp=sharing)
)

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; High-throughput protocol related to the EcoRI RAD-tag procedure above. Optimized for processing 96 samples at one using a liquid handling robot.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Citation: [Tin *et al.* Molecular Ecology Resources 2015, 14:790](http://ecoevo.unit.oist.jp/lab/pubs/Tin,2015,Degenerate%20adaptor%20sequences%20for%20detecting%20PCR%20duplicates%20in%20reduced%20representation%20sequencing%20data%20improve%20genotype%20calling%20accuracy.pdf)


  
* * *

### General protocols

  * [Gel-free size selection and purification of library][10]

Useful for processing large numbers of samples at once.

  * [Picogreen assay][11]

Most accurate measure of overall dsDNA concentration.

 [1]: https://docs.google.com/document/d/1SizyI5bcrbxeGxmE-UbK-GtdmqaVvtGrbmjjYbormKw
 [2]: https://docs.google.com/document/d/1qG-Dv24MyYxz9HQF3ZpWrD43MCI_xPsYsfB-WBlu7Oc
 [3]: https://docs.google.com/document/d/1GW4TzTCJAm8bjqOSC9-fmNRVPbS-3Ij5-2tSLYZygJM
 [4]: http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0096793
 [5]: https://docs.google.com/document/d/1EbyVf2A5pdBlkxBu57tkFkaBiG0lDifdd7FQGcidM4I
 [6]: https://docs.google.com/document/d/1_1amZDvLEIP_sjJzUSELjig3VKI2aALOzr2o5pezAPE/edit
 [7]: http://www.biomedcentral.com/1471-2164/14/790
 [9]: https://docs.google.com/document/d/1I-9MMC2_P9_QcFFHj0vTpoLw06ZeWSXMZiw0YDE9ne0/edit?usp=sharing
 [10]: https://docs.google.com/document/d/163otzQfwyUGtJCNa9Yd7awlsFGE74yWv59LUUNJhqbI/edit?usp=sharing
 [11]: https://docs.google.com/document/d/1ArdwOExyYEy0NtB_u3uIg68VHCBAkU9VbhW7uNENE28/edit?usp=sharing
